### ISSUE TITLE
Remember layout configuration

### ISSUE THREAD

**Issue by dreamflasher** [bigbluebutton/bigbluebutton#16572]
**Is your feature request related to a problem? Please describe.**
Every time I join a bbb video conference (or rejoining if I dropped out), I have to configure the screen to my liking from scratch (hide the presentation, mirror myself, hide the chat).

**Describe the solution you'd like**
When I join a conference I expect the same layout that I last configured. So there should be a cookie being set that stores and restores the whole configuration. 

**Describe alternatives you've considered**
Rethink the default layout configuration:
* people are used to see themselves mirrored, all other conference tools do it like that, make this a default
* start with the presentation minimized, the BBB servers are set up in the way that one apparently needs to set up a default presentation, but it should rather be hidden by default, and only when the initial presentation is being changed the presentation should be shown.
* hide the chat by default and only show it when someone writes (eg. have a look at zoom, they show the message in small as a notification, and the chat is hidden by default)

**Comment by Buda9** [comment]
question for devs, is this possible with dark mode too? cookie for dark mode, perhaps?

**Comment by ffdixon** [comment]
These good suggestions.  The solution is to support client-side storing of preferences, which would be matched with the URL of the BigBlueButton server [this we need to do], and to have all servers give a consistent hostname using a [cluster proxy setup](https://docs.bigbluebutton.org./admin/clusterproxy.html) [this we have done].

**Comment by dreamflasher** [comment]
For me it would be totally fine if I had to configure it per server.

**Comment by schrd** [comment]
You could adjust bbb-html5 setting `public.app.userSettingsStorage` to `local`. This lets BBB store settings in browser localStorage instead of sessionStorage. Ensure to set this to `local` only if 
* you have only a single BBB server
* or you you run multiple behind a loadbalancer like scalelite **and** setup these in a cluster proxy setup

I am not sure if this setting keeps layout settings between sessions but it does keep settings like audio notification when someone joins, dark mode etc

**Comment by dreamflasher** [comment]
Stale bots suck

**Comment by ffdixon** [comment]
Yeah, but it does help us review the older issues to see if they are still issues.  There is a lot of refactoring going on in BigBlueButton 3.0: moving the graphQL to sunset meteor, migration to typescript for the client to add type checking, removal of Kurento to our own media recording, and support for plugins.

The point is after we complete 3.0, we'll have a more solid and extensible platform to address some of the long-term issues that have been outstanding for a while.  

We want to close these issues not because the stalebot closed them, but because we really resolved them in the product by building upon the groundwork in 3.0.

**Comment by dreamflasher** [comment]
stalebots suck.

**Comment by nekohayo** [comment]
Related:
* #24008
* #23496  
(I think the dark theme toggle should be automatic by default on every site)

**Comment by nekohayo** [comment]
I see mentions here of setting `public.app.userSettingsStorage` to `local`, but the reality from a user's perspective is that this server-side opt-in doesn't/can't work, I've never seen layout, dark mode, bandwidth savings settings, etc. being preserved in any of the BBB instances I visit as a user.

Is this supposed to work to those particular settings already?  
If so, why isn't saving user settings to local storage the default behavior? I see no reason why this should be opt-in rather than opt-out. 99% of admins won't know (or care) to enable this, so users are just stuck with this UX problem.

**Comment by dreamflasher** [comment]
Stale bots suck
